tests/MathTest.php

```diff
@@ -110,6 +110,23 @@ public function providerExpressions()
             ['(1+2+3+4- 5)*7/100'],
             ['( 1 + 2 + 3 + 4 - 5 ) * 7 / 100'],
 
+            ['1 && 0'],
+            ['1 && 0 && 1'],
+            ['1 || 0'],
+            ['1 && 0 || 1'],
+
+            ['5 == 3'],
+            ['5 == 5'],
+            ['5 != 3'],
+            ['5 != 5'],
+            ['5 > 3'],
+            ['3 > 5'],
+            ['3 >= 5'],
+            ['3 >= 3'],
+            ['3 < 5'],
+            ['5 < 3'],
+            ['3 <= 5'],
+            ['5 <= 5'],
         ];
```

Comment:
Try adding the following tests:

            ['1 + 5 == 3 + 1'],
            ['1 + 5 == 5 + 1'],
            ['1 + 5 != 3 + 1'],
            ['1 + 5 != 5 + 1'],
            ['1 + 5 > 3 + 1'],
            ['1 + 3 > 5 + 1'],
            ['1 + 3 >= 5 + 1'],
            ['1 + 3 >= 3 + 1'],
            ['1 + 3 < 5 + 1'],
            ['1 + 5 < 3 + 1'],
            ['1 + 3 <= 5 + 1'],
            ['1 + 5 <= 5 + 1'],

I think you will find you will need to fix operator precedence.

Also, I would add additional tests with multiply and divide.  Their precedence is different, so we should make sure our precedence is right.  I would also test () in the same fashion.  Our library should match PHP operator precedence and the best thing about how Alex set up the tests is that we are comparing against PHP itself. Always better to have more tests.